lib/puppet/indirector/file_server.rb

```diff
@@ -51,6 +52,7 @@ def search(request)
     Puppet::FileServing::Fileset.merge(*filesets).collect do |file, base_path|
       inst = model.new(base_path, :relative_path => file)
       inst.links = request.options[:links] if request.options[:links]
+      inst.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
       inst.collect
```

Comment:
unrelated to your PR, but should we be doing `inst.collect(request.options[:source_permissions])` like above?